src/protocol/basics/mul/malicious.rs

```diff
@@ -76,15 +79,20 @@ where
 
     let duplicate_multiply_ctx = ctx.narrow(&Step::DuplicateMultiply);
     let random_constant_ctx = ctx.narrow(&Step::RandomnessForValidation);
+    let b_x = b.x().access_without_downgrade();
+    let b_induced_share = Replicated::new(
+        b_x.left().get_induced_value(),
+        b_x.right().get_induced_value(),
+    );
     let (ab, rab) = try_join(
         a.x().access_without_downgrade().multiply_sparse(
-            b.x().access_without_downgrade(),
+            b_x,
             ctx.semi_honest_context(),
             record_id,
             zeros_at,
         ),
         a.rx().multiply_sparse(
-            b.x().access_without_downgrade(),
+            &b_induced_share,
```

Comment:
See protocol 4.15 in https://eprint.iacr.org/2019/386.pdf.

In Step 4: "Circuit emulation:", it says:

"If Gj is a multiplication gate: Given pairs ([x], [[rb· x]]) and ([y], [[rb· y]]) on the left and right input wires
respectively, the parties compute ([x · y], [[rb· x · y]]) as follows:
(a) The parties call Fmult on [x] and [y] to receive [x · y].
(b) The parties locally compute the induced share [[y]] = f([y], 0, . . . , 0).
(c) The parties call Fbmult on [[rb· x]] and [[y]] to receive [[rb· x · y]].
"